src/Infrastructure/Services/UserService.cs

```diff
@@ -425,73 +428,115 @@ public async Task<Response<string>> RespondToMfaChallengeAsync(string mfaCode, s
 
         private string CleanPhoneNumber(string phoneNumber)
         {
-            // ensure the phone number is in the correct format for cognito ie +441234567890
-            phoneNumber = phoneNumber.Replace(" ", "").Replace("-", "").Replace("(", "").Replace(")", "");
-            if (!phoneNumber.StartsWith("+"))
+            PhoneNumberUtil phoneUtil = PhoneNumberUtil.GetInstance();
+
+            // Ensure it's a valid UK number using libphonenumber
+            var number = phoneUtil.Parse(phoneNumber, "GB");
+            if (!phoneUtil.IsValidNumber(number))
             {
-                phoneNumber = $"+44{phoneNumber.TrimStart('0')}";
+                throw new ValidationException(new List<ValidationFailure>
+                {
+                    new ValidationFailure("PhoneNumber", $"{number} is not a valid UK number")
+                }, "PhoneNumber");
             }
 
+            // Check if the number is specifically a mobile number
+            if (phoneUtil.GetNumberType(number) != PhoneNumberType.MOBILE)
+            {
+                throw new ValidationException(new List<ValidationFailure>
+                {
+                    new ValidationFailure("PhoneNumber", $"{number} is not a valid UK mobile number")
+                }, "PhoneNumber");
+            }
+
+            // Ensure the phone number is in the correct format for cognito i.e., +441234567890 E164
+            phoneNumber = phoneUtil.Format(number, PhoneNumberFormat.E164);
```

Comment:
reusing the input parameter here, just `return phoneUtil.Format(number, PhoneNumberFormat.E164);`